sdks/typescript/src/v1/declaration.ts

```diff
@@ -266,12 +266,52 @@ export class BaseWorkflowDeclaration<
     input: I,
     options?: RunOpts,
     _standaloneTaskName?: string
-  ): Promise<WorkflowRunRef<O>> {
+  ): Promise<WorkflowRunRef<O>>;
+  async runNoWait(
+    input: I[],
+    options?: RunOpts,
+    _standaloneTaskName?: string
+  ): Promise<WorkflowRunRef<O>[]>;
+  async runNoWait(
+    input: I | I[],
+    options?: RunOpts,
+    _standaloneTaskName?: string
+  ): Promise<WorkflowRunRef<O> | WorkflowRunRef<O>[]> {
     if (!this.client) {
       throw UNBOUND_ERR;
     }
 
-    const res = await this.client.admin.runWorkflow<I, O>(this.name, input, options);
+    if (Array.isArray(input)) {
+      let resp: WorkflowRunRef<O>[] = [];
+      for (let i = 0; i < input.length; i += 500) {
+        const batch = input.slice(i, i + 500);
+        const batchResp = await this.client.admin.runWorkflows<I, O>(
+          batch.map((inp) => ({
+            workflowName: this.definition.name,
+            input: inp,
+            options,
+          }))
+        );
+        resp = resp.concat(batchResp);
+      }
```

Comment:
if it's possible, I'd _much_ prefer to do this with `flatMap` and specifying the slice indices beforehand - much less prone to bugs IMO

Reply:
i think we kick this one for now

Reply:
fine with me 👍